index.js

```diff
@@ -1,260 +1,156 @@
-
-    this.es.addEventListener('put', function(e) {
-      if (e && e.data) {
-        _self.features = JSON.parse(e.data);
-        delete _self.disconnected;
-        _self.initialized = true;
-      }
-      cb();
-    });
-
-    this.es.addEventListener('patch', function(e) {
-      if (e && e.data) {
-        try {
-          var patch = JSON.parse(e.data);
-          if (patch && patch.path && patch.data && patch.data.version) {
-            old = pointer.get(_self.features, patch.path);
-            if (old === null || old.version < patch.data.version) {
-              pointer.set(_self.features, patch.path, patch.data);
-            }
-          }
-        } catch(e) {}  // do not update a flag that does not exist
+    update_processor.start(function(err) {
+      if (err) {
+        client.emit('error', err);
       }
-    });
-
-    this.es.addEventListener('delete', function(e) {
-      if (e && e.data) {
-        try {
-          var data = JSON.parse(e.data);
-
-          if (data && data.path && data.version) {
-            old = pointer.get(_self.features, data.path);
-            if (old === null || old.version < data.version) {
-              pointer.set(_self.features, data.path, {
-                "deleted": true,
-                "version": data.version
-              });
-            }
-          }
-        } catch(e) {}  // do not delete a flag that does not exist
+      else if (!initialized) {
+        initialized = true;
+        client.emit('ready');
       }
     });
-
-    this.es.onerror = function(e) {
-      if (e && e.status == 401) {
-        throw new Error("[LaunchDarkly] Invalid API key");
-      }
-      if (!_self.disconnected) {
-        _self.disconnected = new Date().getTime();      
-      }
-    }    
-  }
-
-  client.get_flag = function(key, user, default_val, fn) {
-    client.toggle(key, user, default_val, fn);
+  } else {
+    client.emit('ready');
   }
 
   client.toggle = function(key, user, default_val, fn) {
     sanitize_user(user);
     var cb = fn || noop;
 
-    var request_params = {
-      method: "GET",
-      headers: {
-        'Authorization': 'api_key ' + this.api_key,
-        'User-Agent': 'NodeJSClient/' + VERSION
-      },
-      timeout: this.timeout * 1000,
-      agent: this.proxy_agent
-    };
-
-    var request = make_request(client, '/api/eval/features/' + key);
-
-    if (this.offline) {
+    if (this.is_offline()) {
+      config.logger.info("[LaunchDarkly] toggle called in offline mode. Returning default value.");
       cb(null, default_val);
       return;
     }
 
     else if (!key) {
-      send_flag_event(client, key, user, default_val, default_val);
+      config.logger.error("[LaunchDarkly] No feature flag key specified. Returning default value.");
+      send_flag_event(key, user, default_val, default_val);
       cb(new Error("[LaunchDarkly] No flag key specified in toggle call"), default_val);
       return;
     }
 
     else if (!user) {
```

Comment:
We also should check here to see if the user key exists and return the default if it is not there